### ISSUE TITLE
trouble with pysidecaster.py on windows with anaconda distribution

### ISSUE THREAD

**Issue by geisjb01** [clariusdev/cast#72]
I've been attempting to get pysidecaster.py to work on my windows computer running anaconda. I'm not familiar with running python code in any way other than from an IDE. So far I've made sure to remove the conflicting pycast library. I'm a fair bit lost when it comes to the part where the documentation says to place the .so files in the path.

**Comment by clariusk** [comment]
note our pycast library is built using python version 3.8, so you'll need to ensure you're working with that version, depending on the IDE, you should be able to specify the library source path as well. Alternatively the DLL can reside in your executable path (manually copied), or you can update the PATH environment variable - the latter here is typically a better solution for command line execution.

**Comment by geisjb01** [comment]
Switching back to version 3.8 did help and the issue with the DLL file location seems to have been resolved. However, now the pyside2 module cannot be found even though it was installed with pip.

**Comment by clariusk** [comment]
double check that the pip version is also on 3.8, chance are you may have installed pyside2 (and others) with the default python version that was on your system

**Comment by geisjb01** [comment]
I have confirmed that the pip used is also on 3.8. It is pip 22.1.1 specifically. I also confirmed that the pyside2 package installed on my system and looks like it should be in the path.
![image](https://user-images.githubusercontent.com/106112354/170310226-99102ceb-be29-49f0-a372-730d0c31250b.png)

**Comment by clariusk** [comment]
and what is the error output when executing pysidecaster

**Comment by geisjb01** [comment]
That error has resolved itself. A good system restart resolved the issue with the library. When running the code I connected to the probe. The gui gives a readout saying that I'm connected, but nothing appears other than a black screen.
![image](https://user-images.githubusercontent.com/106112354/170317097-e7c67588-ef44-4867-9250-b12474e66326.png)

**Comment by geisjb01** [comment]
Alrighty, when making the exception for the program what should I look for because there is not an executable for the code. Do I need to make an executable for the program?

**Comment by clariusk** [comment]
right, with python you may not want to unblock the entire program. with the C API the callback provides the UDP port used in streaming (managed by the O/S), but that is not exposed in our python wrapper. so your best solution may be to temporarily disable the firewall and test on a closed network.
i also searched a programmatic way to potentially do this when executing: https://codereview.stackexchange.com/questions/211163/add-and-enable-disable-windows-firewall-rules-with-python

**Comment by geisjb01** [comment]
I first tried the programmatic way (both the methods found in your link and some found in my own digging) and that did not work and it did not appear to be an issue with the programmatic methods not working. Just to be sure I also tried disabling the firewall and testing on a closed network. This was not anymore successful. Thoughts?

**Comment by clariusk** [comment]
typically this is always firewall related, but you could also double check pycaster.py to see if any image callback occurs

**Comment by geisjb01** [comment]
I did not see any callback errors. When getting on the scanners network this morning a new problem has developed where the casting port as displayed by the app is 0. I've tried logging out and logging back in to see if that would fix it as prescribed in one of the other threads with a similar issue.

**Comment by geisjb01** [comment]
With some more fiddling, I have been able to make the example cast an image, but do not see the controls for freezing the image in the api. Is that something I will need to add?

**Comment by clariusk** [comment]
on the python examples, it's pretty basic - so you can add more controls as required, the Qt/C++ programs have some additional controls

**Comment by geisjb01** [comment]
Noted. My next step is to incorporate pulling the IMU data from the scanner. To do this I would simply combine the cast code with the other example for pulling the IMU data, right?

**Comment by clariusk** [comment]
that's correct, thanks

**Comment by geisjb01** [comment]
Thanks for all your help and assistance!

**Comment by ChristophG123T** [comment]
> With some more fiddling, I have been able to make the example cast an image, but do not see the controls for freezing the image in the api. Is that something I will need to add?

Hello, how did you fiddle with this to get it working? I am getting this issue and the programmatic way doesn't work.

**Comment by geisjb01** [comment]
@ChristophG123T I've done a lot of modifications to the code since then so it's a bit fuzzy. I believe what I needed to do was to first connect to the wifi-direct connection on my computer as I would a typical wifi connection. After doing this I completely disabled the firewall entirely because none of the methods I had used to make an exception in the firewall for the program worked. If you need to make an exception instead of disabling everything then you may have some luck compiling the python into an executable and making an exception for that program. Best of luck!
